packages/react/src/date-range-picker/utils.ts

```diff
@@ -0,0 +1,66 @@
+import { isBefore, subMonths, differenceInMonths, closestTo } from 'date-fns';
+
+// If the end date is before the start date, swap the end date with the start
+// This prevents the users from typing invalid date ranges
+export function ensureValidDateRange(dateRange: {
+	from: Date | undefined;
+	to: Date | undefined;
+}) {
+	const { to, from } = dateRange;
+
+	if (from && to && isBefore(to, from)) {
```

Comment:
Should we prevent users from typing (and picking) the same date as well?

Reply:
I don't think so, as I feel like single day date ranges should be valid. What do you think?

Reply:
If we allow it, then the calendar UI should update accordingly, I guess the "selected date" should be fully round to show that the start and end are applied, otherwise it looks broken 
<img width="780" alt="image" src="https://github.com/agriculturegovau/agds-next/assets/853552/e2a54786-be19-4078-8ade-08929831298b">